pkg/es/config/flags.go

```diff
@@ -70,342 +68,283 @@ const (
 	defaultIndexPrefix            = ""
 )
 
-var defaultIndexOptions = config.IndexOptions{
+var defaultIndexOptions = IndexOptions{
 	DateLayout:        initDateLayout(defaultIndexRolloverFrequency, defaultIndexDateSeparator),
 	RolloverFrequency: defaultIndexRolloverFrequency,
 	Shards:            5,
 	Replicas:          1,
 	Priority:          0,
 }
 
-// TODO this should be moved next to config.Configuration struct (maybe ./flags package)
-
-// Options contains various type of Elasticsearch configs and provides the ability
-// to bind them to command line flag and apply overlays, so that some configurations
-// (e.g. archive) may be underspecified and infer the rest of its parameters from primary.
-type Options struct {
-	Primary namespaceConfig `mapstructure:",squash"`
-
-	others map[string]*namespaceConfig
-}
-
-type namespaceConfig struct {
-	config.Configuration `mapstructure:",squash"`
-	namespace            string
-}
-
-// NewOptions creates a new Options struct.
-func NewOptions(primaryNamespace string, otherNamespaces ...string) *Options {
-	// TODO all default values should be defined via cobra flags
-	defaultConfig := DefaultConfig()
-	options := &Options{
-		Primary: namespaceConfig{
-			Configuration: defaultConfig,
-			namespace:     primaryNamespace,
-		},
-		others: make(map[string]*namespaceConfig, len(otherNamespaces)),
-	}
-
-	// Other namespaces need to be explicitly enabled.
-	defaultConfig.Enabled = false
-	for _, namespace := range otherNamespaces {
-		options.others[namespace] = &namespaceConfig{
-			Configuration: defaultConfig,
-			namespace:     namespace,
-		}
-	}
-
-	return options
-}
-
-func (cfg *namespaceConfig) getTLSFlagsConfig() tlscfg.ClientFlagsConfig {
-	return tlscfg.ClientFlagsConfig{
-		Prefix: cfg.namespace,
-	}
-}
-
 // AddFlags adds flags for Options
-func (opt *Options) AddFlags(flagSet *flag.FlagSet) {
-	addFlags(flagSet, &opt.Primary)
-	for _, cfg := range opt.others {
-		addFlags(flagSet, cfg)
-	}
+func (c *Configuration) AddFlags(flagSet *flag.FlagSet) {
+	addFlags(flagSet, c, primaryNamespace)
+	addFlags(flagSet, c, archiveNamespace)
```

Comment:
@yurishkuro wanted to get your thoughts on if this is a good way to initialize the CLI flags for v1?

Reply:
sgtm, except see next comment